src/iarc7_motion/iarc_tasks/track_roomba_task.py

```diff
@@ -91,77 +93,102 @@ def _current_velocity_callback(self, data):
+
+                if not (self._height_checker.above_min_maneuver_height(
+                            self._drone_odometry.pose.pose.position.z)):
+                    return (TaskAborted(msg='Drone is too low'),)
+                elif not (self._z_holder.check_z_error(
+                    self._drone_odometry.pose.pose.position.z)):
+                    return (TaskAborted(msg='Z error is too high'),)
+                    
+
+                try:
+                    roomba_transform = self._tf_buffer.lookup_transform(
+                                        'level_quad',
+                                        self._roomba_id,
+                                        rospy.Time.now(),
+                                        rospy.Duration(self._TRANSFORM_TIMEOUT))
+                except (tf2_ros.LookupException,
+                        tf2_ros.ConnectivityException,
+                        tf2_ros.ExtrapolationException) as ex:
+                    rospy.logerr('ObjectTrackTask: Exception when looking up transform')
+                    rospy.logerr(ex.message)
+                    return (TaskAborted(msg='Exception when looking up transform during roomba track'),)
+
+                # Creat point centered at drone's center
+                stamped_point = PointStamped()
+                stamped_point.point.x = 0
+                stamped_point.point.y = 0
+                stamped_point.point.z = 0
+
+                # returns point distances of roomba to center point of level quad
+                self._roomba_point = tf2_geometry_msgs.do_transform_point(
+                                                    stamped_point, roomba_transform)
+
+                if not self._check_max_start_roomba_range():
+                    return (TaskAborted(msg='The provided roomba is not found or not within a meter of the quad'),)
+
+                if self._check_max_ending_roomba_range():
+                    return (TaskDone(),)
+
+                # p-controller
+                x_vel_target = (self._roomba_point.point.x * self._K_X + 
+                            self._roomba_odometry.twist.twist.linear.x)
+                y_vel_target = (self._roomba_point.point.y * self._K_Y + 
+                            self._roomba_odometry.twist.twist.linear.y)
+                z_vel_target = self._z_holder.get_height_hold_response(
+                    self._drone_odometry.pose.pose.position.z)
+
+                #caps velocity
+                vel_target = math.sqrt(x_vel_target**2 + y_vel_target**2)
+
+                if vel_target > self._MAX_TRANSLATION_SPEED:
+                    x_vel_target = x_vel_target * (self._MAX_TRANSLATION_SPEED/vel_target)
+                    y_vel_target = y_vel_target * (self._MAX_TRANSLATION_SPEED/vel_target)
+                
+                if (abs(z_vel_target) > self._MAX_Z_VELOCITY):
+                    z_vel_target = z_vel_target/abs(z_vel_target) * self._MAX_Z_VELOCITY
+
+                desired_vel = []
```

Comment:
Can we just make the whole list in one shot (i.e. `desired_vel = [x_vel_target, y_vel_...]`)?